src/goto-programs/add_race_assertions.cpp

```diff
@@ -15,18 +15,22 @@ class w_guardst
 
   std::list<irep_idt> w_guards;
 
-  const symbolt &
-  get_guard_symbol(const irep_idt &object, const exprt &original_expr)
+  const symbolt &get_guard_symbol(
+    const irep_idt &object,
+    const exprt &original_expr,
+    bool deref)
   {
-    const irep_idt identifier = "tmp_" + id2string(object);
+    const irep_idt identifier =
+      deref ? "_ESBMC_deref_" + id2string(object) : "tmp_" + id2string(object);
```

Comment:
Just for my own sanity: would it be possible to use 2 underscores, i.e., `__ESBMC_deref_` as prefix? I don't think we're using a single underscore (and we shouldn't, because implementation stuff is supposed to use two).

Also, maybe for a later PR, I think we have a similar technique in place already in the value_set: it uses a suffix, though, something like `[]` for arrays and `.member` for structs/unions. It could be that you're replicating a similar strategy here, in which case it might make sense to unify both of them... Not for this PR, though.